### PR TITLE
Add address tab in ManageWhitelist modal

### LINKED ISSUE #3177: [UI] Implement Add address content in Manage Whitelist modal
## Specification

Implement content of Add address tab in ManageWhitelist modal, try to reuse existing upload/input logic

### Design (optional, might not be applicable)
* https://www.figma.com/file/Phby7kEYihhvGgQn69A314/Verified-Recipients?node-id=81%3A20508
* https://www.figma.com/file/Phby7kEYihhvGgQn69A314/Verified-Recipients?node-id=83%3A24856

### PR DESCRIPTION
**Pull request by danbr**
## Description

-This pr adds content to the whitelisted tab, it allows addresses to be verified and submitted.
-It refactors `UploadAddressesWidget` into many parts to be more generic and allow use in its current location `Whitelist`, and support the new `ManageWhitelist` dialog.

-As the supporting logic is not ready, I use the same saga `updateWhitelist` for submitting.
-To allow addresses to be added, whitelist extension needs to be installed.

New stuff ✨

-`UploadAddresses` component
-`WhitelistedValidation` schemas

Single address input:
Screenshot 2022-03-10 at 15 38 08
Screenshot 2022-03-10 at 15 38 39

File input:
Screenshot 2022-03-10 at 15 38 22
Screenshot 2022-03-10 at 15 39 05

Resolves #3177

### CODE REVIEW

**Comment by alicjakujawa** [comment]
@danbr For the description: we dont need to have whitelist extension installed to be able to add addresses to the "whitelist". Those concepts are totally separated.

File: src/modules/core/components/CSVUploader/CSVUploader.tsx [hunk 1]
```diff
@@ -73,11 +80,12 @@ const CSVUploader = ({
         accept: MIME_TYPES,
       }}
       customErrorMessage={error}
-      inputStatusAppearance={{ theme: 'minimal', textSpace: 'wrap' }}
+      inputStatusAppearance={{ theme: 'fat', textSpace: 'wrap' }}
```
Comment (alicjakujawa) [inline src/modules/core/components/CSVUploader/CSVUploader.tsx:83]: Are we sure we want change it for all cases?

File: src/modules/dashboard/components/Dialogs/ManageWhitelistDialog/ManageWhitelistDialogForm.tsx [hunk 2]
```diff
@@ -44,27 +45,48 @@ const MSG = defineMessages({
...
-
   const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
-
   const hasRegisteredProfile = !!username && !ethereal;
-
   const userHasPermission = hasRegisteredProfile && hasRoot(allUserRoles);
 
+  const { isWhitelistExtensionEnabled } = useEnabledExtensions({
+    colonyAddress: colony.colonyAddress,
+  });
```
Comment (alicjakujawa) [inline src/modules/dashboard/components/Dialogs/ManageWhitelistDialog/ManageWhitelistDialogForm.tsx:88]: we dont need this

File: src/modules/dashboard/components/Whitelist/WhitelistAddresses/UploadAddressesExtensionWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+              disabled={!userHasPermission || !isValid || isSubmitting}
+              type="submit"
+              loading={isSubmitting}
+            />
+          </div>
+        </>
+      )}
+    </ActionForm>
+  );
+};
```
Comment (alicjakujawa) [inline src/modules/dashboard/components/Whitelist/WhitelistAddresses/UploadAddressesExtensionWidget.tsx:119]: If we will be using this component in both cases, its need to be adjusted to fit this new "whitelist concept"

**Review by alicjakujawa — CHANGES_REQUESTED** [review]

**Review by arrenv — CHANGES_REQUESTED** [review]
@danbr to add to Alicja's comments. This feature will be storing the whitelisted addresses in IPFS, similar to how a colony stores it's tokens in the manage tokens functionality.

File: src/modules/core/components/CSVUploader/CSVUploader.tsx (hunk 1 above)
Comment (danbr) [reply src/modules/core/components/CSVUploader/CSVUploader.tsx:83]: @alicjakujawa This changes only for the 2 cases where addresses for Whitelisting can be provided, inside the Whitelist Extension page, and here in ManageWhitelist modal. Therefore, i wanted the look and feel to be consistent. Besides which, the coloured dot was not always present with 'minimal' theme.

File: src/modules/dashboard/components/Dialogs/ManageWhitelistDialog/ManageWhitelistDialogForm.tsx (hunk 2 above)
Comment (danbr) [reply src/modules/dashboard/components/Dialogs/ManageWhitelistDialog/ManageWhitelistDialogForm.tsx:88]: @alicjakujawa no problem........ i'll simply remove the check.

File: src/modules/dashboard/components/Whitelist/WhitelistAddresses/UploadAddressesExtensionWidget.tsx (hunk 3 above)
Comment (danbr) [reply src/modules/dashboard/components/Whitelist/WhitelistAddresses/UploadAddressesExtensionWidget.tsx:119]: We will be using `UploadAddresses` in both situations. 
Perhaps I am not aware of all the details of the new "whitelist concept", but `UploadAddresses` should support both Whitelisting. If there is some functionality missing, or if you foresee any issues, please let me know.

**Comment by danbr** [comment]
> @danbr to add to Alicja's comments. This feature will be storing the whitelisted addresses in IPFS, similar to how a colony stores it's tokens in the manage tokens functionality.

@arrenv @alicjakujawa I am a little confused, what you are describing sounds like issue #3179 (Implement Add address logic). Although out of the list of tasks for #3179, i think most are already handled here, but not storing on IPFS.

- Implement/Reuse upload logic
- It should handle both - add addresses by upload OR input
- Store whitelist addresses on IPFS in a colony's Metadata (like list of colony tokens)
- Show proper validation errors

To avoid being blocked, I used the existing saga for submitting the addresses as a way of testing the form validation and displaying errors & success messages, while waiting for issue #3179 to be completed.

Should "Store whitelist addresses on IPFS in a colony's Metadata (like list of colony tokens)" now be handled within this PR?

**Comment by arrenv** [comment]
> Should "Store whitelist addresses on IPFS in a colony's Metadata (like list of colony tokens)" now be handled within this PR?

No, it shouldn't be included in this PR, I think that having to install the Extensions plugin though creates a link that should not be there for this feature though.

**Comment by danbr** [comment]
> 

@arrenv Very good. I have already removed whitelist extension check.

**Review by arrenv — APPROVED** [review]
The UI is looking to be showing correctly for me.

I noticed a couple of minor UI issues:
- 'Download template' shifts up by 1px when you select a file.
- The underline under the tabs is showing differently based on the different heights of the dialog window

02582158-88eb-43ac-9d02-6bb411d4d8ec

**Review by chinins — APPROVED** [review]
Other than minor UI things Arren noted, all good!

**Comment by danbr** [comment]
> Other than minor UI things Arren noted, all good!

> @chinins @arrenv "I noticed a couple of minor UI issues:"
> 'Download` template' shifts up by 1px when you select a file.

-I fixed the height of the drag'n'drop/browse area, so regardless of if a file is present, the sizes should remain constant.

Screenshot 2022-03-16 at 15 55 46
Screenshot 2022-03-16 at 16 00 06


> The underline under the tabs is showing differently based on the different heights of the dialog window

I do not have an issue with this. How did you change the heights of the dialog window?
Assuming the 'selected tab' line should be above the base line (NOT how it is designed in Figma where it is on top), for me it does not have any space in between, and looks fine.

On my system
Screenshot 2022-03-16 at 16 39 24

Figma
Screenshot 2022-03-16 at 17 00 37

@alicjakujawa's colonyDapp#3218 also looks the same on my system, does it also have an issue on your system?
Screenshot 2022-03-16 at 17 18 03



If there is still an issue, and as it is outside of the scope of this PR, perhaps @alicjakujawa could investigate?

**Review by alicjakujawa — APPROVED** [review]
Approving this. I will create bug for fixing tabs border alignment and problem with modal content  so no worry of those here :)

**Review by arrenv — APPROVED** [review]
Approving this based on @alicjakujawa newly created issue.

**Pull request merged**